tests/jballerina-unit-test/src/test/resources/test-src/types/intersection/test_intersection_type_negative.bal

```diff
@@ -43,3 +43,24 @@ type Z record {
     X x;
     Y y;
 };
+
+type ReadonlyType readonly;
+type FutureType future<int>;
+type T2 FutureType & ReadonlyType;
+
+function testReadonlyIntersectionImmutability() {
+    ReadonlyType & string[] arr = ["a", "b", "c"];
+    arr.push("d");
+}
+
+type Foo record {
+    string p1;
+    int p2?;
+};
+
+function testAssigningMutableValuesToReadonlyIntersections() {
+    int[] numArray = [1, 2, 3, 4];
+    ReadonlyType & int[] readonlyNumArray = numArray;
+    Foo foo = {p1: "P1"};
+    Foo & ReadonlyType readonlyFoo = foo;
```

Comment:
```suggestion
    Foo & ReadonlyType _ = foo;
```

Update wherever applicable.

Reply:
Added in [commit](https://github.com/ballerina-platform/ballerina-lang/pull/42025/commits/495a8a56919ce85ea13a6d6dfad1690268d44c09).